### DISCUSSION TITLE
How to Integrate it

### DISCUSSION THREAD

**Discussion by ToKie-code** [osk2/yamaha-soundbar#10] [Q&A] [answered]
I have copied all files from yamaha_soundbar GitHub to custom_components/yamaha_soundbar (because I could not find the yamaha_soundbar on HACS) and added this at the end of the configuration.yaml 
media_player:
  - platform: yamaha_soundbar
    host: 192.168.69.68
    name: YamahaSRX40A # To name your sources (optional)
    sources:
      {
        'HDMI': 'TV', 
        'optical': 'Plexamp', 
        'bluetooth': 'Bluetooth',
      }
and I got this error when I check the Config
Platform error 'media_player' from integration 'yamaha_soundbar' - Integration 'yamaha_soundbar' not found.
Sorry I'm a newbie at Home Assistant, YAML and Python . I'm fine with C# and other ;-)

**Comment by Perlfaucon** [comment, marked as answer]
Did you copied just the files ? Because when you extract the .zip it created many "yamaha_soundbar".
For me that created 2 yamaha-soundbar-master and in the second I have "custom_components" and the files to copie for HA is this "yamaha_soundbar"
![Capture d'écran 2024-12-05 121653](https://github.com/user-attachments/assets/9ebdbefd-31b1-41b3-84f9-2240546f1884)

Be sure to copie that file, for exemple:  

H:\Bureau\yamaha-soundbar-master\yamaha-soundbar-master\custom_components

In the last custom_components.
